src/kafka-util/src/client.rs

```diff
@@ -280,8 +308,80 @@ where
+
+        match rewrites.get(&addr) {
+            None => {
+                match &self.default_override {
+                    Some(default_tunnel) => {
+                        let broker = addr.clone();
+                        let (tunnel_id, default_tunnel) = default_tunnel.clone();
+                        let ssh_tunnel_manager = self.ssh_tunnel_manager.clone();
+                        let runtime = self.runtime.clone();
+                        // We spawn an additional thread here, and immediately
+                        // join it so that the rdkafka thread calling this callback
+                        // spends all its time parked, instead of being
+                        // continuously woken up when we `block_on` the
+                        // ssh tunnel creation.
+                        //
+                        // Note that we hold the async lock while we create this tunnel; this is to
+                        // prevent competing threads from thrashing the ssh bastion host.
+                        let ssh_tunnel = match std::thread::spawn(move || {
+                            runtime.block_on(async move {
+                                ssh_tunnel_manager
+                                    .connect(
+                                        tunnel_id,
+                                        default_tunnel,
+                                        &broker.host,
+                                        broker.port.parse().unwrap(),
+                                    )
+                                    .await
+                            })
+                        })
+                        .join()
+                        {
+                            Ok(ssh_tunnel) => ssh_tunnel,
+                            Err(e) => std::panic::resume_unwind(e),
+                        };
+
+                        match ssh_tunnel {
+                            Ok(ssh_tunnel) => {
+                                let rewrite_tunnel = ssh_tunnel.clone();
+                                rewrites.insert(
+                                    addr.clone(),
+                                    Arc::new(move || {
+                                        let addr = rewrite_tunnel.local_addr();
+                                        BrokerRewrite {
+                                            host: addr.ip().to_string(),
+                                            port: Some(addr.port()),
+                                        }
+                                    }),
+                                );
+
+                                let addr = ssh_tunnel.local_addr();
+                                BrokerAddr {
+                                    host: addr.ip().to_string(),
+                                    port: addr.port().to_string(),
+                                }
+                            }
+                            Err(e) => {
+                                // Ensure a log line shows up.
```

Comment:
Does this cause the error to get reported to the user via `mz_source_status_history`? Because if it's just for our logs, seems simpler + cleaner to just emit via a `warn!` here directly.

Reply:
thats a good point! I will change this to a warn